src/turbopuffer.ts

```diff
@@ -300,13 +300,24 @@ export class Namespace {
    * Fetches the approximate number of vectors in a namespace.
    */
   async approxNumVectors(): Promise<number> {
-    const response = await this.client.http.doRequest<object>({
+    return (await this.metadata()).approx_count;
+  }
+
+  async metadata(): Promise<NamespaceMetadata> {
+    const response = await this.client.http.doRequest<NamespaceMetadata>({
       method: "HEAD",
       path: `/v1/vectors/${this.id}`,
       retryable: true,
     });
-    const num = response.headers.get("X-turbopuffer-Approx-Num-Vectors");
-    return num ? parseInt(num) : 0;
+
+    return {
+      id: this.id,
+      approx_count: parseInt(
+        response.headers.get("X-turbopuffer-Approx-Num-Vectors")!,
+      ),
+      dimensions: parseInt(response.headers.get("X-turbopuffer-Dimensions")!),
+      created_at: new Date(response.headers.get("X-turbopuffer-Created-At")!),
```

Comment:
is there a way we can test that this parsing actually works? I'm assuming it's super loosey goosy, didn't realize new Date() had a string constructor like that... JS is weird

Reply:
Great point @morgangallant! Made the test a bit more verbose here